code/game/objects/structures/training_machine.dm

```diff
@@ -0,0 +1,310 @@
+		return FALSE
+
+/obj/structure/training_machine/post_buckle_mob()
+	handle_density()
+	return ..()
+
+/obj/structure/training_machine/post_unbuckle_mob()
+	handle_density()
+	return ..()
+
+/obj/structure/training_machine/emag_act(mob/user)
+	. = ..()
+	if (obj_flags & EMAGGED)
+		return
+	obj_flags |= EMAGGED
+	remove_attached_item(throwing = TRUE) //Toss out the old attached item!
+	attach_item(new /obj/item/storage/toolbox/syndicate(src))
+	to_chat(user, "<span class='warning'>You override the training machine's safety protocols, and activate its realistic combat feature. A toolbox pops out of a slot on the top.</span>")
+	playsound(src, 'sound/machines/click.ogg', 50, TRUE)
+	add_overlay("evil_trainer")
+
+/obj/structure/training_machine/examine(mob/user)
+	. = ..()
+	if (obj_flags & EMAGGED)
+		. += "<span class='warning'>It has a dangerous-looking toolbox attached to it, and the control panel is smoking sightly...</span>"
+	else if (attached_item) //Can't removed the syndicate toolbox!
+		. += "<span class='notice'><b>Alt-Click to remove \the [attached_item]</b></span>"
+	. += "<span class='notice'><b>Click to open control interface.</b></span>"
+
+/**
+  * Device that simply counts the number of times you've hit a mob or target with. Looks like a toolbox but isn't.
+  *
+  * Also has a 'Lap' function for keeping track of hits made at a certain point. Also, looks kinda like his grace for laughs and pranks.
+  */
+/obj/item/training_toolbox
+	name = "Training Toolbox"
+	desc = "AURUMILL-Brand Baby's First Training Toolbox. A digital display on the back keeps track of hits made by the user. Second toolbox sold seperately!"
+	icon_state = "his_grace_ascended"
+	inhand_icon_state = "toolbox_gold"
+	lefthand_file = 'icons/mob/inhands/equipment/toolbox_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/equipment/toolbox_righthand.dmi'
+	flags_1 = CONDUCT_1
+	force = 0
+	throwforce = 0
+	throw_speed = 2
+	throw_range = 7
+	w_class = WEIGHT_CLASS_BULKY
+	///Total number of hits made against a valid target
+	var/total_hits = 0
+	///Number of hits made since the Lap button (alt-click) was last pushed
+	var/lap_hits = 0
+
+/obj/item/training_toolbox/afterattack(atom/target, mob/user, proximity)
+	. = ..()
+	if (!proximity || target == user)
+		return
+	if (check_hit(target))
+		user.changeNext_move(CLICK_CD_MELEE)
+
+/obj/item/training_toolbox/proc/check_hit(atom/target)
+	var/target_is_machine = istype(target, /obj/structure/training_machine)
+	if (!ismob(target) && !istype(target, /obj/item/target) && !target_is_machine)
+		return FALSE
+	if (target_is_machine)
+		var/obj/structure/training_machine/trainer = target
+		if (!trainer.attached_item)
+			return FALSE
+	total_hits++
+	lap_hits++
+	playsound(src,'sound/weapons/smash.ogg',50,FALSE)
+	if (lap_hits % HITS_TO_KILL == 0)
+		playsound(src,'sound/machines/twobeep.ogg',25,FALSE)
+	return TRUE
+
+/obj/item/training_toolbox/throw_impact(atom/hit_atom, datum/thrownthing/throwingdatum)
+	..()
```

Comment:
```suggestion
	. = ..()
```

Reply:
Actually, wouldn't it be better to have
```
if (!..())
  check_hit(hit_atom)
```
so things like getting caught by another mob won't trigger it?

Reply:
If you do that then use this pattern:
```cpp
. = ..()
if(!.)
```
Always preserve the parent call's return unless you have a very good reason not to.